packages/repository/src/relations/relation.helpers.ts

```diff
@@ -151,10 +151,7 @@ function isInclusionAllowed<T extends Entity, Relations extends object = {}>(
  * @param targetKey - name of the target key
  *
  */
-export function flattenTargetsOfOneToOneRelation<
-  SourceWithRelations extends Entity,
-  Target extends Entity
->(
+export function flattenTargetsOfOneToOneRelation<Target extends Entity>(
```

Comment:
Removing the first of two parameters can break existing code, we must be careful here. 

I assume most places in our codebase are leveraging type inference provided by the compiler, so no change is needed there. The places passing two parameters (`SourceWithRelations`, `Target`) will be found by the compiler and trigger compilation errors, so that should be easy to handle too.

However: is `flattenTargetsOfOneToOneRelation` a part of our public API? If yes, then we need to communicate this change to our users.

Reply:
I did search references of `flattenTargetsOfOneToOneRelation` and it's only used internally. There is no mentioning in our docs either. Please also note that even our code does not explicitly provide the generic parameter types.

For this case, I think it's fine to fix it without bearing the extra overhead.